tools/integration_tests/readonly/list_objects_test.go

```diff
@@ -60,15 +65,20 @@ func TestListObjectsInBucketSubDirectory(t *testing.T) {
 		log.Fatal(err)
 	}
 
-	if len(obj) != NumberOfObjectsInTestBucketSubDirectory {
+	// Comparing number of objects in the directory of testBucket - 2.
+	if len(obj) != NumberOfObjectsInDirectoryTestBucket {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != FileInSubDirectoryNameInTestBucket && obj[0].IsDir() != false {
-		t.Errorf("Listed object is incorrect.")
+	// Comparing first object name and type.
+	// Name - testBucket/Test/a.txt, Type - File
+	if obj[0].Name() != FileNameInDirectoryTestBucket || obj[0].IsDir() != false {
+		t.Errorf("Listed object from bucket directory is incorrect.")
```

Comment:
nit: Object Listed for file in bucket directory is incorrect.

Reply:
fixed